specs/agents/tracing-distributed-tracing.md

```diff
@@ -55,10 +114,55 @@ Agents MUST implement these validation rules when mutating `tracestate`:
 - Vendor values have a maximum size of 256 chars.
 - Vendor values may only contain ASCII RFC0020 characters (i.e., the range `0x20` to `0x7E`) except comma `,` and `=`.
 - In addition to the above limitations, the keys and values used in the `es` entry must not contain the characters `:` and `;`.
-- If adding another key/value pair to the `es` entry would exceed the limit of 256 chars,
-  that key/value pair MUST be ignored by agents.
-  The key/value and entry separators `:` and `;` have to be considered as well.
+- If adding another key/value pair to the `es` entry would exceed the limit of
+  256 chars (including separator characters `:` and `;`), that key/value pair
+  MUST be ignored by agents.
 
 Note that we will currently only ever populate an `es` `tracestate` entry at the trace root.
 It is not strictly necessary to validate `tracestate` in its entirety when received downstream.
 Instead, downstream agents may opt to only parse the `es` entry and skip validation of other vendors' entries.
+This means that the vendor key validations are only relevant if an agent adds
+its own non-`es` keys to tracestate
+
+In addition, we do not enforce the 32-entry limit for vendor entries in
+`tracestate`. Doing so would cripple our ability to use `tracestate` for our
+own purposes, arbitrarily. Removing other entries to make way for our own
+would also cause unexpected behavior. In any case, this situation should be
+rare and we feel comfortable ignoring the validation rules in this case.
+
+
+### HTTP Headers
+
+Every outgoing request should be intercepted and modified to include both the
+`traceparent` and `tracestate` headers, described above.
+
+If an incoming request contains either of the `traceparent` or `tracestate`
+headers, they should be propagated throughout the transaction and mutated as
+specified above before being set on outgoing requests.
+
```

Comment:
Do we think it's worth calling out the need to handle multiple `tracestate` headers [per the W3C spec](https://www.w3.org/TR/trace-context/)?

> Tracestate MAY be sent or received as multiple header fields. Multiple tracestate header fields MUST be handled as specified by RFC7230 Section 3.2.2 Field Order.

Reply:
I'm not certain it's necessary. [HTTP allows for duplicate headers](https://stackoverflow.com/a/4371395/1370767). Most implementations in python combine multiple headers into a single entry that's comma-separated already, so it's something we end up just not having to deal with. We provide [an abstraction](https://github.com/elastic/apm-agent-python/blob/87b4de97558479aeeced651bc023c417c3630c11/elasticapm/utils/disttracing.py#L108-L127) that eliminates the need to deal with these duplicate headers (and can be overwritten for any libraries which deal with them differently, such as `aiohttp` in python).